universe/logic/api/auth.go

```diff
@@ -83,6 +85,38 @@ func VerifyPolkadotSignature(wallet, challenge, signature string) (bool, error)
 	return pub.Verify(sig, transcript)
 }
 
+func signHash(data []byte) []byte {
+	msg := fmt.Sprintf("\x19Ethereum Signed Message:\n%d%s", len(data), data)
+	return crypto.Keccak256([]byte(msg))
+}
+
+func VerifyEthereumSignature(address, challenge, signature string) (bool, error) {
```

Comment:
Perhaps add datatypes for address and challenge as you did for signature

Reply:
Yeah my IDE suggests the same, even though i personally like to give everything a type..
To me its more consistent

Reply:
If a programming language is trying to make our life easier by writing a cleaner and more compact code, we should let it, no? :)

Reply:
I noticed it's used in the codebase, that's why I also did - I probably wouldn't dare otherwise :)